docs/source/en/model_doc/vits.md

```diff
@@ -0,0 +1,40 @@
+<!--Copyright 2023 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# VITS
+
+## Overview
+
+The VITS model was proposed in [Conditional Variational Autoencoder with Adversarial Learning for End-to-End Text-to-Speech](https://arxiv.org/abs/2106.06103) by Jaehyeon Kim, Jungil Kong, Juhee Son.
+
+The abstract from the paper is the following:
+
+*Several recent end-to-end text-to-speech (TTS) models enabling single-stage training and parallel sampling have been proposed, but their sample quality does not match that of two-stage TTS systems. In this work, we present a parallel end-to-end TTS method that generates more natural sounding audio than current two-stage models. Our method adopts variational inference augmented with normalizing flows and an adversarial training process, which improves the expressive power of generative modeling. We also propose a stochastic duration predictor to synthesize speech with diverse rhythms from input text. With the uncertainty modeling over latent variables and the stochastic duration predictor, our method expresses the natural one-to-many relationship in which a text input can be spoken in multiple ways with different pitches and rhythms. A subjective human evaluation (mean opinion score, or MOS) on the LJ Speech, a single speaker dataset, shows that our method outperforms the best publicly available TTS systems and achieves a MOS comparable to ground truth.*
+
+This model can also be used with MMS-TTS checkpoints as those use the same architecture (but a different tokenizer).
+
+This model was contributed by [Matthijs](https://huggingface.co/Matthijs). The original code can be found [here](https://github.com/jaywalnut310/vits).
+
+## VitsConfig
+
+[[autodoc]] VitsConfig
+
+## VitsTokenizer
+
+[[autodoc]] VitsTokenizer
+    - __call__
+    - save_vocabulary
+
+## VitsModel
+
+[[autodoc]] VitsModel
+    - forward
```

Comment:
(inferring from Matthijs's design!)

VITS **only** works as a TTS model, hence adding `ForTextToSpeech` is redundant in this case, since this is a given. The naming follows the same convention as other models in the lib, where if there's only one task, we don't append it to the end of the model name.

E.g. CLAP only works as a joint text-audio embedding extraction model, it doesn't serve any other tasks. Thus, it's named as `ClapModel` in transformers:
https://github.com/huggingface/transformers/blob/f3e96235a324dfd14a3e4ed4535259768b6c9b0e/src/transformers/models/clap/modeling_clap.py#L1938
and not `ClapModelForEmbeddingExtraction`.

Whereas SpeechT5 serves a range of tasks (ASR / Voice Cloning / TTS) -> here we have to append the task on the end of the model name to differentiate between them.

Happy to go the other way if you feel strongly that `VitsForTextToSpeech` gives a more unified API between models

Reply:
OK. Let's stick with `VitsModel`. 

For CLAP I think the reasoning is a little different. All of the CLIP-like models follow this pattern as we don't have feature extraction as a defined task and we can see the returned embeddings as "raw" outputs that can have other model heads built on-top. 

In this case, it's more similar to SAM (which just has SamModel). This model does produce outputs for a specific task, it's just that there isn't really a "head" and it only does one downstream task.